feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/OrchestratorFragment.kt

```diff
@@ -83,6 +81,11 @@ internal class OrchestratorFragment : Fragment(R.layout.fragment_orchestrator) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        if (savedInstanceState != null) {
+            // Only updating when explicitly restoring the state
+            orchestratorVm.requestProcessed = savedInstanceState.getBoolean(KEY_REQUEST_PROCESSED)
```

Comment:
If the flag is moved out of the Fragment, then we don't need to save and restore it anymore?

Reply:
Hmm, I'd argue that this makes it more complex and error-prone. Previously, the request itself (in the args) and the request status were kept inside the Fragment. So, if you just ensure that status is persisted through Fragment restoration, you ensure that the same request won't be handled twice.
Now, a central place keeps the status but if we have two Fragment instances which request is the status referring to?

Reply:
Afaik, the view model is not shared between fragment instances in the case of the parallel flow. Although due to the other fix it should not be an issue anymore.

Reply:
After some poking around, this helps when restoring the screen after minimisation. I would rather keep the flag stored in a saved state for now. At least until we figure out the correct way to restore the execution flow.

Reply:
Oh, right! FSR I was thinking of the Orchestrator as a singleton while yeah, it's an instance connected to the particular Activity/Fragment pair.